src/frontend/src/monitor/stats.rs

```diff
@@ -80,3 +83,77 @@ impl FrontendMetrics {
         GLOBAL_FRONTEND_METRICS.clone()
     }
 }
+
+#[derive(Clone)]
+pub struct CursorMetrics {
+    pub valid_subsription_cursor_nums: IntGauge,
+    pub invalid_subsription_cursor_nums: IntGauge,
+    pub subscription_cursor_error_count: GenericCounter<AtomicU64>,
+    pub subscription_cursor_query_duration: HistogramVec,
+    pub subscription_cursor_declare_duration: HistogramVec,
+    pub subscription_cursor_fetch_duration: HistogramVec,
+    pub subscription_cursor_last_fetch_duration: HistogramVec,
```

Comment:
I think it is simpler and less error prone if we use `Collector` to periodically report the following metrics instead of updating them in cursor manager in an ad-hoc manner:
- `valid_subsription_cursor_nums`
- `invalid_subsription_cursor_nums`
- `subscription_cursor_last_fetch_duration`

You can refer to the [StateStoreCollector ](https://github.com/risingwavelabs/risingwave/blob/746850f12673f6f87300893abd4839d288453042/src/storage/src/monitor/hummock_state_store_metrics.rs#L582)as an exmaple.